### PR TITLE
[PNUDEV-18] Banners implemented

### PR DESCRIPTION
**Pull request by Igor-Polatajko**
### Admin panel

**Banners display**

image

**Edit banner**

image

**Banner preview**

image

**Save banner updates**

image

### API

**Only active banners are present in API response**

image

image

image

#### Note: To add new banner, the only thing you need to change is `BannerType` enum (changes will be automatically picked up).

image

### CODE REVIEW

File: src/main/java/com/pnu/dev/radioserviceapi/mongo/Banner.java [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+@Document
+@Data
+@Builder(toBuilder = true)
+@NoArgsConstructor
+@AllArgsConstructor
+public class Banner {
+
+    @Id
+    private BannerType bannerType;
```
Comment (volodymyrvynnyk) [inline src/main/java/com/pnu/dev/radioserviceapi/mongo/Banner.java:18]: Great idea!

File: src/main/java/com/pnu/dev/radioserviceapi/service/BannerApiService.java [hunk 2]
```diff
@@ -0,0 +1,12 @@
+package com.pnu.dev.radioserviceapi.service;
+
+import org.springframework.stereotype.Service;
+
+import java.util.Map;
+
+@Service
```
Comment (volodymyrvynnyk) [inline src/main/java/com/pnu/dev/radioserviceapi/service/BannerApiService.java:7]: No needs in this annotation here

File: src/main/java/com/pnu/dev/radioserviceapi/service/BannerApiServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+    @Autowired
+    public BannerApiServiceImpl(BannerRepository bannerRepository) {
+        this.bannerRepository = bannerRepository;
+    }
+
+    @Override
+    public Map<String, String> findBannersToShow() {
+        return bannerRepository.findAll().stream()
+                .filter(Banner::isShow)
```
Comment (volodymyrvynnyk) [inline src/main/java/com/pnu/dev/radioserviceapi/service/BannerApiServiceImpl.java:24]: Or add `findAllByIsShow(boolean show)` method to Repository, but let it be

File: src/main/java/com/pnu/dev/radioserviceapi/service/BannerServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+        return bannerRepository.findById(bannerType)
+                .orElseGet(() -> {
+                    Banner banner = Banner.builder()
+                            .bannerType(bannerType)
+                            .markdown(StringUtils.EMPTY)
+                            .build();
+
+                    bannerRepository.save(banner);
+                    return banner;
+                });
```
Comment (volodymyrvynnyk) [inline src/main/java/com/pnu/dev/radioserviceapi/service/BannerServiceImpl.java:37]: Hardly readable

File: src/main/resources/templates/banner/edit.ftl [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+
+        <div class="row d-flex justify-content-between px-4 pt-4">
+            <div class="custom-control custom-checkbox">
+                <input type="checkbox" class="custom-control-input" id="primary-banner-checkbox"
+                       name="show" ${banner.show?string('checked','')}>
+                <label class="custom-control-label" for="primary-banner-checkbox">Показувати банер</label>
+            </div>
+            <button class="btn btn-primary">Зберегти</button>
+        </div>
+    </form>
```
Comment (volodymyrvynnyk) [inline src/main/resources/templates/banner/edit.ftl:19]: As I merged Security, add pls hidden csrf input

File: src/main/java/com/pnu/dev/radioserviceapi/service/BannerApiService.java (hunk 2 above)
Comment (Igor-Polatajko) [reply src/main/java/com/pnu/dev/radioserviceapi/service/BannerApiService.java:7]: Thank you for being so attentive, I missed it.

File: src/main/java/com/pnu/dev/radioserviceapi/service/BannerApiServiceImpl.java (hunk 3 above)
Comment (Igor-Polatajko) [reply src/main/java/com/pnu/dev/radioserviceapi/service/BannerApiServiceImpl.java:24]: Yes, sure. 
`List<Banner> findAllByShowTrue();` would be even better

**Pull request merged**